hub/core/dataset/dataset.py

```diff
@@ -1303,3 +1342,187 @@ def append(self, sample: Dict[str, Any], skip_ok: bool = False):
                                 "Error while attepting to rollback appends"
                             ) from e2
                     raise e
+
+    def _view_hash(self):
+        return hashlib.sha1(
+            (
+                f"{self.path}[{':'.join(str(e.value) for e in self.index.values)}]@{self.version_state['commit_id']}&{getattr(self, '_query', None)}"
+            ).encode()
+        ).hexdigest()
+
+    def _get_view_info(self):
+        if not hasattr(self, "_view_info"):
+            tm = getattr(self, "_created_at", time())
+            hash = self._view_hash()
+            info = {
+                "id": hash,
+                "description": "Virtual Datasource",
+                "virtual-datasource": True,
+                "source-dataset": self.path,
+                "source-dataset-version": self.version_state["commit_id"],
+                "created_at": tm,
+            }
+
+            query = getattr(self, "_query", None)
+            if query:
+                info["query"] = query
+                info["source-dataset-index"] = getattr(self, "_source_ds_idx", None)
+            self._view_info = info
+        return self._view_info
+
+    @staticmethod
+    def _write_queries_json(ds, info):
```

Comment:
When a view is saved in the `.queries` sub directory of a dataset (either the source dataset itself, or `hub://username/queries` if source ds is read only), its info is written to a  `queries.json` file in ds storage in addition to the vds's info.

```
Source ds storage
    |
    |-----queries.json
    |-----.queries
                |
                |-----vds1
                |-----vds2
                |-----vds3
```